src/nats.h

```diff
@@ -4564,6 +4564,28 @@ natsSubscription_AutoUnsubscribe(natsSubscription *sub, int max);
 NATS_EXTERN natsStatus
 natsSubscription_QueuedMsgs(natsSubscription *sub, uint64_t *queuedMsgs);
 
+/** \brief Gets the subscription id.
+ *
+ * Returns the id of the given subscription.
+ *
+ * @param sub the pointer to the #natsSubscription object.
+ * @param id if not `NULL`, the memory location where to store the subscription id.
+ */
+NATS_EXTERN natsStatus
+natsSubscription_GetID(natsSubscription* sub, int64_t* id);
+
+/** \brief Gets the subject name.
+ *
+ * Returns the subject of the given subscription.
+ *
+ * \warning The string belongs to the subscription and must not be freed. Copy it if needed.
+ *
+ * @param sub the pointer to the #natsSubscription object.
+ * @param subject if not `NULL`, the memory location where to store the subject name.
```

Comment:
Same comment that for the ID. Now, as a convention, we could say that those getter return "invalid" values that the user would have to interpret as the subscription being closed/invalid. For instance, natsSubscription_GetID() could be returning a int64_t (instead of status), and assume that 0 means invalid/closed. And natsSubscription_GetSubject() could return NULL to indicate invalid/closed subscription (since empty subject is not valid).